src/Microsoft.Performance.Toolkit.Engine/EngineCreateInfo.cs

```diff
@@ -14,53 +17,72 @@ namespace Microsoft.Performance.Toolkit.Engine
+        /// </param>
+        /// <remarks>
+        ///     If no extension directory is specified, the current directory is used.
+        /// </remarks>
+        /// <exception cref="InvalidExtensionDirectoryException">
+        ///     Thrown if <paramref name="extensionDirectory"/> is an invalid directory path.
+        /// </exception>
+        public EngineCreateInfo(string extensionDirectory = null)
+            : this(new string[] { extensionDirectory ?? Environment.CurrentDirectory })
         {
-            this.ExtensionDirectory = Environment.CurrentDirectory;
         }
 
         /// <summary>
-        ///     Gets or sets the extension directory from
-        ///     which the runtime instance is to load plugins.
-        ///     By default, this value is the current working
-        ///     directory. If you set it to <c>null</c>,
-        ///     the current working directory will be used.
+        ///     Initializes a new instance of the <see cref="EngineCreateInfo"/> class.
         /// </summary>
-        public string ExtensionDirectory
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown is <paramref name="extensionDirectories"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="InvalidExtensionDirectoryException">
+        ///     Thrown if <paramref name="extensionDirectories"/> contains an invalid directory path.
+        /// </exception>
+        public EngineCreateInfo(IEnumerable<string> extensionDirectories)
         {
-            get
-            {
-                return this.extensionDirectory;
-            }
-            set
+            Guard.NotNull(extensionDirectories, nameof(extensionDirectories));
+
+            var directories = new List<string>();
+
+            foreach (var directory in extensionDirectories)
             {
-                var candidateValue = value ?? Environment.CurrentDirectory;
+                if (string.IsNullOrWhiteSpace(directory))
+                {
+                    throw new InvalidExtensionDirectoryException(directory);
+                }
 
                 DirectoryInfo dirInfo;
                 try
                 {
-                    dirInfo = new DirectoryInfo(candidateValue);
+                    dirInfo = new DirectoryInfo(directory);
                 }
                 catch (Exception e)
                 {
-                    throw new InvalidExtensionDirectoryException(candidateValue, e);
+                    throw new InvalidExtensionDirectoryException(directory, e);
                 }
 
                 if (!dirInfo.Exists)
                 {
-                    throw new InvalidExtensionDirectoryException(value);
+                    throw new InvalidExtensionDirectoryException(directory);
                 }
 
-                this.extensionDirectory = dirInfo.FullName;
+                directories.Add(dirInfo.FullName);
             }
+
+            this.ExtensionDirectories = directories.AsReadOnly();
```

Comment:
Does it make sense to dedupe this list here?

Reply:
Zack and I spoke offline. This makes sense, but DirectoryInfo doesn't implement IEquatable or similar and making this work across platforms would be challenging. We're just going to leave this for now.